src/client/client.ts

```diff
@@ -1009,35 +1056,39 @@ export default class TaskClient {
       async () => {
         let ruConsumption = 0;
 
-        await Array.from(Object.values(sprocs)).map(async ({ id, fn }) => {
-          try {
-            const result = await this._client.replaceSproc(id, fn);
-            ruConsumption += result.ruConsumption || 0;
-          } catch (err) {
-            // 404 means we need to create it
-            if (IronTaskError.is(err, ErrorCode.DATABASE_RESOURCE_NOT_FOUND)) {
-              try {
-                const result = await this._client.createSproc(id, fn);
-                ruConsumption += result.ruConsumption || 0;
-              } catch (err) {
-                // 409s are okay at this point. Just means we
-                // were in a race with something else
-                if (
-                  IronTaskError.is(
-                    err,
-                    ErrorCode.DATABASE_RESOURCE_ALREADY_EXISTS
-                  )
-                ) {
-                  return;
+        await Promise.all(
+          await Array.from(Object.values(sprocs)).map(async ({ id, fn }) => {
```

Comment:
shouldn't need the `await` on here any more

```suggestion
          Array.from(Object.values(sprocs)).map(async ({ id, fn }) => {
```